clap_complete/src/macros.rs

```diff
@@ -2,7 +2,13 @@ macro_rules! w {
     ($buf:expr, $to_w:expr) => {
         match $buf.write_all($to_w) {
             Ok(..) => (),
-            Err(..) => panic!("Failed to write to generated file"),
+            Err(err) if err.kind() == std::io::ErrorKind::BrokenPipe => {
+                // Is there a better exit code for this?
+                std::process::exit(1);
+            }
+            Err(err) => {
+                panic!("Failed to write to generated file: {0}", err);
+            }
```

Comment:
Neither case should panic but we should report this error up.  Please open an issue on this for us to track for our next breaking change.

Reply:
Opened #5457.
Should I close this?

Reply:
The solution will change and some time past, so sure.